packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2851,10 +2852,13 @@ def reset_state(
         :param restart_from_round: from which round to restart the abci. This round should be the first round in the last period.
         :param round_count: the round count at the beginning of the period -1.
         :param reset_index: the reset index (a.k.a. period count) -1.
+        :param serialized_db_state: the state of the database at the beginning of the period. If provided, the database will be reset to this state.
         """
         self._reset_to_default_params()
         self.abci_app.synchronized_data.db.round_count = round_count
         self.abci_app.reset_index = reset_index
+        if serialized_db_state is not None:
+            self.abci_app.synchronized_data.db.sync(serialized_db_state)
```

Comment:
Small coverage gap here. Let's cover it on #1776 to save CI time as I also need to fill a coverage gap there.

Reply:
https://github.com/valory-xyz/open-autonomy/pull/1775#issuecomment-1408962700